app/handler/books.go

```diff
@@ -112,3 +128,17 @@ func DeleteBook(w http.ResponseWriter, r *http.Request) {
 	books = newSetOfBooks
 	json.NewEncoder(w).Encode(books)
 }
+
+func findOrFail(id string) (book Book, err error) {
+	for _, b := range books {
+		if b.ID == id {
+			book = b
+		}
+	}
+
+	if book.ID == "" {
+		err = fmt.Errorf("Book Not Found")
```

Comment:
don't assign the error, return it, `return nil, err`

Reply:
True but when I `return nil, err` it complains that `have (nil, error)` `wants (Book, err)`
So I just `return book, err` cause book here will be `<nil>`